time[] start_times
time end_time